### PR TITLE
Combine two object_access_hook into one

### PR DESCRIPTION
**Pull request by fanfuxiaoran**

### CODE REVIEW

File: diskquota_utility.c [hunk 1]
```diff
@@ -110,6 +108,7 @@ static bool   to_delete_quota(QuotaType type, int64 quota_limit_mb, float4 segra
 static void   check_role(Oid roleoid, char *rolname, int64 quota_limit_mb);
 
 List *get_rel_oid_list(void);
+void  diskquota_stop_worker(void);
```
Comment (beeender) [inline diskquota_utility.c:111]: this is declared in header file, no need to be declared here. Actually I think compiler will report a warning.

File: gp_activetable.c [hunk 2]
```diff
@@ -181,12 +181,15 @@ object_access_hook_QuotaStmt(ObjectAccessType access, Oid classId, Oid objectId,
 	if (prev_object_access_hook) (*prev_object_access_hook)(access, classId, objectId, subId, arg);
 
 	// if is 'drop extension diskquota'
-	if (classId == ExtensionRelationId)
+	if (classId == ExtensionRelationId && access == OAT_DROP)
```
Comment (beeender) [inline gp_activetable.c:184]: the function is doing much more than `active_table`  in this file. Shall we consider moving the hook to an independent file, and expose methods from `gp_activetable` for better code organizing?

But it is not the scope of this PR.

**Review by beeender — COMMENTED** [review]
Nice! Code deletion!!

Please don't merge this before the 2.0.0 release.

File: diskquota_utility.c (hunk 1 above)
Comment (fanfuxiaoran) [reply diskquota_utility.c:111]: OK!

File: gp_activetable.c (hunk 2 above)
Comment (fanfuxiaoran) [reply gp_activetable.c:184]: Agree. We can put all the hooks into one separated file.

**Review by beeender — APPROVED** [review]
Merge after `2.0.0` release

File: gp_activetable.c [hunk 3]
```diff
@@ -188,10 +188,7 @@ object_access_hook_QuotaStmt(ObjectAccessType access, Oid classId, Oid objectId,
 			invalidate_database_rejectmap(MyDatabaseId);
 		}
 
-		if (IS_QUERY_DISPATCHER())
```
Comment (xuebinsu) [inline gp_activetable.c]: Why removing this? I don't think we have any worker on segments

**Review by xuebinsu — COMMENTED** [review]
LGTM overall

File: gp_activetable.c (hunk 3 above)
Comment (fanfuxiaoran) [reply gp_activetable.c]:
```
update_diskquota_db_list(MyDatabaseId, HASH_REMOVE);
```
This is called in diskquota_stop_worker, which will clean monitored db cache on all segments.

Comment (xuebinsu) [reply gp_activetable.c]: Then should we move `update_diskquota_db_list()` outside of `diskquota_stop_worker()`?

Comment (fanfuxiaoran) [reply gp_activetable.c]: I don't think so. It's good to call update_diskquota_db_list() in diskquota_stop_worker().

**Review by Sasasu — APPROVED** [review]
`diskquota_stop_worker` also remote the dbid cache. fine but a little multi-usage function.

File: gp_activetable.c (hunk 3 above)
Comment (beeender) [reply gp_activetable.c]: `IS_QUERY_DISPATCHER()` was added back? cannot see the change anymore :confused:

**Pull request merged**